pkg/rc/farm.go

```diff
@@ -65,22 +67,27 @@ func NewFarm(
 	logger logging.Logger,
 	rcSelector klabels.Selector,
 	alerter alerting.Alerter,
+	errorReporter error_reporter.Reporter,
```

Comment:
alternatively I could make the constructor place a no-op reporter on the struct and then add a SetErrorReporter function. that might be a little nicer than spreading `nil` everywhere